includes/class-wc-stripe-webhook-handler.php

```diff
@@ -525,15 +525,33 @@ public function process_webhook_capture( $notification ) {
 	 * @param object $notification
 	 */
 	public function process_webhook_charge_succeeded( $notification ) {
-		// The following payment methods are synchronous so does not need to be handle via webhook.
-		if ( ( isset( $notification->data->object->source->type ) && WC_Stripe_Payment_Methods::CARD === $notification->data->object->source->type ) || ( isset( $notification->data->object->source->type ) && 'three_d_secure' === $notification->data->object->source->type ) ) {
+		if ( empty( $notification->data->object ) ) {
+			WC_Stripe_Logger::log( 'Missing charge object in charge.succeeded webhook' );
 			return;
 		}
 
-		$order = WC_Stripe_Helper::get_order_by_charge_id( $notification->data->object->id );
+		// https://docs.stripe.com/api/events/types#event_types-charge.succeeded
+		$charge = $notification->data->object;
+
+		// The following payment methods are synchronous so does not need to be handled via webhook.
+		$payment_method_type = $this->get_payment_method_type_from_charge( $charge );
+		if (
+			in_array(
+				$payment_method_type,
+				[
+					WC_Stripe_Payment_Methods::CARD,
+					WC_Stripe_Payment_Methods::AMAZON_PAY,
+					'three_d_secure',
+				],
+				true
+			) ) {
+			return;
+		}
```

Comment:
(nit) Moving the array out of the condition makes it slightly more readable.


```suggestion
		$synchronous_methods = [
            WC_Stripe_Payment_Methods::CARD,
            WC_Stripe_Payment_Methods::AMAZON_PAY,
            'three_d_secure',
        ];
        
        if ( in_array( $payment_method_type, $synchronous_methods, true ) ) {
            return;
        }
```